cmake/Modules/FindMKLDNN.cmake

```diff
@@ -28,6 +28,11 @@ IF(NOT MKLDNN_FOUND)
     endif()
 
     set(DNNL_MAKE_COMMAND "cmake" "--build" ".")
+    include(ProcessorCount)
+    ProcessorCount(proc_cnt)
+    if ((DEFINED ENV{MAX_JOBS}) AND ("$ENV{MAX_JOBS}" LESS_EQUAL ${proc_cnt}))
+      list(APPEND DNNL_MAKE_COMMAND "-j" "$ENV{MAX_JOBS}")
```

Comment:
`-l` is `make` and `ninja` option. Unfortunately I don't see it for `cmake --build .` which is used here. Do you know how to pass it in such a case?

Alternatively, we can consider to switch `BUILD_COMMAND` from `cmake` to `make` or current `CMAKE_GENERATOR`, but this will require handling generator options separately which can be non-trivial with possible expansion to Windows build. For the reference, here in other pytorch script make is hardcoded it seems with both `-j` and `-l`:
https://github.com/pytorch/pytorch/blob/7530cfe7e41dfab26f678c56843e97f128508c17/cmake/External/nccl.cmake#L37 

```
$ cmake --build
Usage: cmake --build <dir>             [options] [-- [native-options]]
       cmake --build --preset <preset> [options] [-- [native-options]]
Options:
  <dir>          = Project binary directory to be built.
  --preset <preset>, --preset=<preset>
                 = Specify a build preset.
  --list-presets
                 = List available build presets.
  --parallel [<jobs>], -j [<jobs>]
                 = Build in parallel using the given number of jobs.
                   If <jobs> is omitted the native build tool's
                   default number is used.
                   The CMAKE_BUILD_PARALLEL_LEVEL environment variable
                   specifies a default parallel level when this option
                   is not given.
  --target <tgt>..., -t <tgt>...
                 = Build <tgt> instead of default targets.
  --config <cfg> = For multi-configuration tools, choose <cfg>.
  --clean-first  = Build target 'clean' first, then build.
                   (To clean only, use --target 'clean'.)
  --verbose, -v  = Enable verbose output - if supported - including
                   the build commands to be executed.
  --             = Pass remaining options to the native tool.
```

Reply:
Actually I probably can use `-- -l N` to pass this to native tool. Let me try.

Reply:
Ok, added `-l N` in case cmake generator is of Make or Ninja family.